src/averageAges.js

```diff
@@ -55,7 +85,35 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const findRelative = function(momName, array) {
+    return array.find(el => el.name === momName);
+  };
+
+  const findRelatives = people.reduce((acc, el) => {
+    const mother = findRelative(el.mother, people);
+
+    if (mother) {
+      return [...acc, {
+        ...el, motherBirth: mother.born,
+      }];
+    }
+
+    return acc;
+  }, []);
+
+  const findRelativesSon = findRelatives.filter(el => el.sex === 'm');
+
```

Comment:
Merge those iterations
```suggestion
  const filteredSons = people.reduce((acc, person) => {
    const mother = findRelative(person.mother, people);

    if (mother && person.sex === 'm') {
      return [...acc, {
        ...person, motherBirth: mother.born,
      }];
    }

    return acc;
  }, []);
```